### PR TITLE
fix: support maps in step parameters

### PR DESCRIPTION
**Pull request by jayatalr**
*Issue #, if available:*

*Description of changes:* This change is made to support map types in step properties.

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ x ] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [ x ] I certify that the changes I am introducing will be backword compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [ x ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ x ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ x ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ x ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ x ] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [ x ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ x ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/workflow/properties.py [hunk 1]
```diff
@@ -109,6 +111,39 @@ def __getitem__(self, item: Union[int, str]):
...
+class PropertiesMap(Properties):
+    """PropertiesMap for use in workflow expressions."""
+
+    def __init__(self, path: str, shape_name: str = None):
+        """Create a PropertiesMap instance representing the given shape.
+
+        Args:
+            path (str): The parent path of the PropertiesMap instance.
+            shape_name (str): The botocore sagemaker service model shape name.
+            root_shape_name (str): The botocore sagemaker service model shape name.
```
Comment (dpatro) [inline src/sagemaker/workflow/properties.py:123]: `root_shape_name` is not present in the arguments. We should remove this from doc

File: tests/integ/test_workflow.py [hunk 2]
```diff
@@ -855,6 +855,113 @@ def test_one_step_callback_pipeline(sagemaker_session, role, pipeline_name, regi
             pass
 
 
+def test_steps_with_map_params_pipeline(
```
Comment (staubhp) [inline tests/integ/test_workflow.py:858]: I might be missing it but how is this testing map params?

Comment (jayatalr) [reply tests/integ/test_workflow.py:858]: assert at line 947 and 948 are checking hyperparameter values? Do we need different asserts?

File: src/sagemaker/workflow/properties.py (hunk 1 above)
Comment (jayatalr) [reply src/sagemaker/workflow/properties.py]: Removed

File: tests/integ/test_workflow.py (hunk 2 above)
Comment (jayatalr) [reply tests/integ/test_workflow.py:858]: Updated for testing hyperparameters

**Review by staubhp — APPROVED** [review]

**Review by dpatro — APPROVED** [review]

**Review by shreyapandit — COMMENTED** [review]
LGTM

**Review by ahsan-z-khan — APPROVED** [review]

**Pull request merged**